app/scripts/views/mixins/signin-mixin.js

```diff
@@ -145,21 +147,22 @@ define(function (require, exports, module) {
         .then(this.navigate.bind(this, this.model.get('redirectTo') || 'settings', {}, navigateData));
     },
 
-    _engageSignInForm (event) {
-      if (event && event.target) {
-        var target = this.$el.find(event.target).attr('id');
-
-        if (target === 'have-account') {
-          // if the user clicks on 'have-account' we count that as flow event instead of the 'engage' event.
-          // Details: https://github.com/mozilla/fxa-content-server/pull/4221
-          this.logEvent('flow.have-account');
-          return;
-        }
+    _clickLink (event) {
+      if (event && event.target && event.target.id) {
+        // Each link on the sign-in form emits a different event,
+        // named according to its id.
+        this.logFlowEvent(event.target.id, this.viewName);
       }
+    },
 
-      // user has engaged with the sign in, sign up or force auth form
-      // the flow event will be different depending on the view name
-      this.logEventOnce(`flow.${this.viewName}.engage`);
-    }
+    _engageSignInForm () {
```

Comment:
Summary of the outcomes from RL discussions about this with @shane-tomlinson and @vbudhram:

1. Create a new `flow-event-mixin`, containing everything but the flow begin event from `flow-begin-mixin`.

2. Move these flow events into said new mixin.

3. Leave the `flow-begin-mixin` as just emitting the flow begin event and nothing else.

Later on, for #4004, we can use the `flow-event-mixin` in the password reset views.